File: api/src/config/conexao_banco.py
import os
import sqlite3

def dict_factory(cursor, row):
    d = {}
    for idx, col in enumerate(cursor.description):
        d[col[0]] = row[idx]
    return d

def obter_conexao():
    caminho_base = os.path.dirname(os.path.abspath(__file__))
    pasta_db = os.path.abspath(os.path.join(caminho_base, '../../db'))
    os.makedirs(pasta_db, exist_ok=True)
    caminho_banco = os.path.join(pasta_db, 'landing.db')

    conexao = sqlite3.connect(caminho_banco)
    conexao.row_factory = dict_factory
    conexao.execute('PRAGMA journal_mode = WAL')
    conexao.execute('PRAGMA foreign_keys = ON')
    return conexao

File: api/src/config/conexaoBanco.js
const { DatabaseSync } = require('node:sqlite')
const path = require('path')
const fs = require('fs')
require('dotenv').config({ path: path.resolve(__dirname, '../../.env') })

// Garante que a pasta db/ existe e define o caminho do banco SQLite
const pastaDb = path.resolve(__dirname, '../../db')
if (!fs.existsSync(pastaDb)) fs.mkdirSync(pastaDb, { recursive: true })
const caminhoBanco = path.join(pastaDb, 'landing.db')

// Cria conexão síncrona com o banco SQLite
const banco = new DatabaseSync(caminhoBanco)

// Ativa WAL para melhor performance em leitura concorrente
banco.exec('PRAGMA journal_mode = WAL')
banco.exec('PRAGMA foreign_keys = ON')

module.exports = banco
